[garywongkai] iP

garywongkai opened the pull request:
# Gluti project

  

![Chatbot Mascot](https://github.com/garywongkai/ip/blob/branch-A-CodingStandard/src/main/data/Gluti.png)

  

```

Welcome to the Gluti chatbot!
```

## **Gluti is your chat task manager, it is:**

* Text-based
* User-friendly
* Easy and Fast to use


## Setup

 1.  Download the release [here!](https://github.com/garywongkai/ip/releases)
 2.  Launch it!
 3.  Add your tasks
 4. ~~????~~ Profit
 
## Features
 - [X] Task Manager
 - [X] Manages Deadlines!
 - [ ] GUI

  

## UML diagrams

 
```mermaid

classDiagram

FileStorage <-- Gluti : Initalises

FileStorage --* Ui

Ui <-- Gluti : Calls

Parser-- FileStorage : Link

Ui --> Parser : Calls

Parser --> Todo

Parser --> Deadline

Parser --> Event

Task --|> Todo : Inheritance

Task--|> Deadline: Inheritance

Task--|> Event: Inheritance

class Gluti{ main()

}

class Ui{ boolean isExit

run()}

class Parser{FileStorage fstorage

boolean isExit

parse(String)

}

class FileStorage{ ArrayList<Task> taskList

checkifexist()

read(String)

saveList(taskList)

readList()

}

class Task {

protected String description

protected boolean isDone

getStatusIcon()

toString()

setDone()

setunDone()

isMatch(keyword)

}

class Todo {

@Override toString()

}

  

class Deadline {

protected String by

validDate(String)

}

class Event {

String[] date;

}

```

  

## Setting up in Intellij

  

Prerequisites: JDK 11, update Intellij to the most recent version.

  

1. Open Intellij (if you are not in the welcome screen, click `File` > `Close Project` to close the existing project first)

  

1. Open the project into Intellij as follows:

  

1. Click `Open`.

  

1. Select the project directory, and click `OK`.

  

1. If there are any further prompts, accept the defaults.

  

1. Configure the project to use **JDK 11** (not other versions) as explained in [here](https://www.jetbrains.com/help/idea/sdk.html#set-up-jdk).<br>

  

In the same dialog, set the **Project language level** field to the `SDK default` option.

  

3. After that, locate the `src/main/java/Duke.java` file, right-click it, and choose `Run Duke.main()` (if the code editor is showing compile errors, try restarting the IDE). If the setup is correct, you should see something like the below as the output:

## The code 🔥
```
static void main(String[] args) {
	//Insert amazing Gluti code here!
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

garywongkai committed:
Level-0

garywongkai committed:
Refactor

garywongkai committed:
Checkpoint

garywongkai committed:
Checkpoint

garywongkai committed:
New Checkpoint

garywongkai committed:
Checkpoint 2

garywongkai committed:
Level-7

garywongkai committed:
Level-7

garywongkai committed:
Checkpoint 3

garywongkai committed:
Checkpoint 3

garywongkai committed:
LocalDate addition

garywongkai committed:
More OOP Checkpoint

garywongkai committed:
Merge branch 'add-gradle-support'

garywongkai committed:
Gradle Test

garywongkai committed:
Checkpoint Gradle Test

garywongkai committed:
New Commit after Testing

garywongkai committed:
Update header comments

garywongkai committed:
Update README.md

garywongkai committed:
Update README.md

garywongkai committed:
Update README.md

garywongkai committed:
Add Find function

garywongkai committed:
Merge branch 'branch-A-CodingStandard'

garywongkai committed:
Merge branch 'branch-Level-9'

garywongkai committed:
Update Branch

garywongkai committed:
Merge branch 'branch-A-JavaDoc'

garywongkai committed:
Update README.md

garywongkai committed:
Implement GUI

garywongkai committed:
Implement GUI

garywongkai committed:
WIP GUI

garywongkai committed:
New Checkpoint

cliftonchee commented on src/main/java/Gluti/helpers/FileStorage.java:
```diff
@@ -0,0 +1,132 @@
+     * Checks and creates the directory and file if there is no such directory or file
+     * @throws GlutiException IO Errors
+     */
+    private void checkifexist() throws GlutiException {
```
Can consider changing to checkIfExist() instead!

cliftonchee commented on src/main/java/Gluti/helpers/Parser.java:
```diff
@@ -0,0 +1,160 @@
+                    }
+                    break;
+                case "todo":
+                    String[] tempinput = word.split(" ", 2);
```
Can consider camelCase for naming convention!

cliftonchee commented on src/main/java/Gluti/helpers/Ui.java:
```diff
@@ -0,0 +1,36 @@
+ */
+public class Ui {
+    private Parser parser;
+    private boolean isExit ;
```
Maybe can consider hasExited instead!

cliftonchee commented on src/main/java/Gluti/utils/Deadline.java:
```diff
@@ -0,0 +1,52 @@
+     * Checks if the input is a valid date and perform the relevant formatting if valid
+     * @param by the due date input
+     */
+    private void validDate(String by) {
```
Can consider renaming by to be "by_date" for further clarity!

cliftonchee commented on src/main/java/Gluti/helpers/Parser.java:
```diff
@@ -0,0 +1,160 @@
+                    }
+                    break;
+                case "event":
+                    String[] tempinpute = word.split(" ", 2);
```
Same as above, or alternatively eventTempInput for further clarity!

cliftonchee reviewed (commented):
Looks good to me! Other than minor naming conventions, very readable code and good work!

garywongkai committed:
GUI Checkpoint

garywongkai committed:
GUI v1

garywongkai committed:
New GUI

garywongkai committed:
GUI Notification

ImgBotApp committed:
[ImgBot] Optimize images

*Total -- 2,462.57kb -> 1,837.90kb (25.37%)

/src/main/resources/data/user.png -- 985.25kb -> 705.23kb (28.42%)
/src/main/resources/data/Gluticon.png -- 1,365.20kb -> 1,022.72kb (25.09%)
/src/main/data/Gluti.png -- 112.12kb -> 109.94kb (1.94%)

Signed-off-by: ImgBotApp <ImgBotHelp@gmail.com>

garywongkai committed:
Minor fixes

garywongkai committed:
Add Assertions to multiple existing methods
Using assertions will allow the code to document important assumptions
at various points

garywongkai committed:
Improve Code Readability
Refactor unnecessary codes and import statements, tidied up code
structure
Applied SLAP and KISS to code quality

garywongkai committed:
Improve codequality and Add Assertions

garywongkai committed:
Merge pull request #1 from garywongkai/imgbot

[ImgBot] Optimize images

garywongkai committed:
Set up Github actions
Continuous Integration implemented to allow for consistent testing of
segments for code. Merged PR with image optimizer from github's imagebot

garywongkai committed:
Merge branch 'master' of https://github.com/garywongkai/ip

starsia commented on src/main/java/Gluti/utils/Deadline.java:
```diff
@@ -0,0 +1,47 @@
+            LocalDate parsedDate = LocalDate.parse(temp, inputFormatter);
+            String formattedDate = parsedDate.format(outputFormatter);
+            this.by_date = " " +formattedDate;
+        } catch (Exception e) {
+            // Handle the parsing exception (invalid format)
+            //System.out.println("Error: Invalid date format");
+        }
+        }
```
Should this bracket be pushed back?

starsia commented on src/main/java/Gluti/utils/Task.java:
```diff
@@ -0,0 +1,29 @@
+    public String toString() {
+        return "[" + getStatusIcon() + "] " + description;
+    }
+    public void setDone() {
+        isDone = true;
+    }
+
+    public void setunDone() { isDone = false; }
```
```suggestion
    public void setDone() {
        isDone = true;
    }

    public void setUndone() { isDone = false; }
```

Would this be what you meant?

starsia commented on src/test/java/Gluti/utils/TodoTest.java:
```diff
@@ -0,0 +1,13 @@
+
+public class TodoTest {
+    @Test
+    public void todoTest(){
+        Todo testTodo = new Todo("todoTest");
+        assertEquals(testTodo.toString(), "[T][ ] todoTest");
```
Would a better name for method be: toString_simpleDescriptionAndUndone_success()?
As per: featureUnderTest_testScenario_expectedBehavior()

starsia commented on src/main/java/Gluti/Gluti.java:
```diff
@@ -0,0 +1,36 @@
+        parser = new Parser(fStorage);
+        ui = new Ui();
+    }
+    public static void main(String[] args) throws GlutiException, IOException {
+        new Gluti();
+    }
```
I like how neat your Main is!

starsia commented on src/main/java/Gluti/helpers/FileStorage.java:
```diff
@@ -0,0 +1,136 @@
+
+    public Task read(String line) {
+        assert line != null : "Next line read should not be empty";
+        char tasktype = line.charAt(1);
```
Should your variable name be camelCase instead of nocase?
```suggestion
        char taskType = line.charAt(1);
```

starsia reviewed (commented):
Java coding standards are mostly adhered to! Great job on the work so far 😃

garywongkai committed:
Add Natural Dates

garywongkai committed:
Update User Guide

garywongkai committed:
UG update

garywongkai committed:
Jar Export
